src/words/environment/WordsObject.java

```diff
@@ -13,13 +15,35 @@
 	private Position cell;
 	private String currentMessage;
 	private Action lastAction;
-
+	private HashMap<WordsObject, ArrayList<Property>> referers;
```

Comment:
I guess this could really be a LinkedList or even just a Set, since the order is not relevant.